### ISSUE TITLE
Add code comments

### ISSUE THREAD

**Issue by oliviertassinari** [mui/mui-toolpad#417]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Make it possible to comment what each API/component/piece of logic is about.

### Examples 🌈

- https://manual.bubble.io/help-guides/maintaining-an-application/commenting

### Motivation 🔦

In a pro-code codebase, you are expected to leave comments on what complex pieces are doing. Why not the same in low-code? Note that this issue is miles different from #416!

**Comment by Janpot** [comment]
Good idea, currently it's already possible to add comments in bindings (before, adding them after results in errors)

![Screenshot 2022-05-15 at 17 40 32](https://user-images.githubusercontent.com/2109932/168481265-852243d1-86b1-450d-89b9-4f79529f997d.png)
